tests/baselines/reference/conditionalOperatorConditionIsBooleanType.types

```diff
@@ -286,7 +286,7 @@ var resultIsString3 = null === undefined ? exprString1 : exprString2;
 var resultIsObject3 = true || false ? exprIsObject1 : exprIsObject2;
 >resultIsObject3 : Object
 >true || false ? exprIsObject1 : exprIsObject2 : Object
->true || false : boolean
+>true || false : true

```

Comment:
This is the wrong result under `strictNullChecks: false`

Reply:
Could u ELI5 how to reason about `strictNullChecks: false` in situations like this?

Reply:
Any value can be falsy due to being `null` / `undefined`

Reply:
I know that from the assignability PoV but that doesnt help me explain your expected result here. Shouldnt intrinsic value like `true` be non-falsy? 

Where is the line if the referenced issue got labeled as a bug?

Reply:
A value of type `true` might be inhabited by `null`. Obviously the expression `true` itself isn't, but that's not really under analysis

Reply:
If the compiler has to always account for the fact that the value might be inhabited by `null` then wouldn't it mean that one of the examples from the referenced issue shouldn't typecheck (and it does typecheck today)?
```ts
type Result = { success: true }
            | { success: false, error: string }

function handleError2(res: Result) {
    if (res.success !== false) {
        return;
    }

    res.error; // OK, but shouldn't be?
}
```

I always thought that `strictNullChecks: false` means that things can be `null`/`undefined` at runtime but that the type-checker is meant to ignore this fact completely.

Reply:
Either way - I pushed out an improved (but not cleaned up, code-wise) version of the fix for the linked issue. Please take a look if those semantics look any better.

Reply:
> I always thought that strictNullChecks: false means that things can be null/undefined at runtime but that the type-checker is meant to ignore this fact completely.

This is a point of discussion that comes up sometimes on Discord - `strictNullChecks: false` is basically a completely different type checker.  The flag does a lot more than its name would suggest, changing inferences, narrowing, etc. in sometimes very unexpected ways, and all of it is more or less by design.